src/lib/Libtpp/tpp_client.c

```diff
@@ -697,7 +697,7 @@ tpp_init(struct tpp_config *cnf)
 	 */
 
 	/* for unix, set a pthread_atfork handler */
-	if (pthread_atfork(NULL, NULL, tpp_terminate)) {
+	if (pthread_atfork(tpp_atfork_prepare, tpp_atfork_parent, tpp_terminate)) {
```

Comment:
I feel we should register the atfork handlers before we call tpp_transport_init() (ie, before tpp starts the threads), ie, before line 646, could be even much much earlier as well - dont see any harm